tests/devapps/XForms/XForms/App.xaml.cs

```diff
@@ -78,15 +79,16 @@ public static void InitPublicClient()
 
             else
             {
-                 // Let Android set its own redirect uri
+                // Let Android set its own redirect uri
                 switch (Device.RuntimePlatform)
                 {
-                    case "iOS":
-                        builder.WithRedirectUri(RedirectUriOnIos);
-                        break;
-                    case "Android":
-                        builder.WithRedirectUri(RedirectUriOnAndroid);
-                        break;
+                case "iOS":
+                    builder.WithRedirectUri(RedirectUriOnIos);
```

Comment:
Although I believe this works, you should generally assume that builders are immutable. This means that when you invoke a "with" method on a builder, the method will return you a new builder that has all the config from the old one plus the new config. 

```
Builder WithRedirectUri(string u)
{
     Builder newBuilder = Builder.Clone();
     newBuilder.RedirectUri = u;
     return newBuilder;
}

// similar to how string is immutable
string s = "hello world";
s.ToUpper();
print s; // it's still "hello world"
```

So it's safer to do 

```csharp
builder = builder.WithRedirectUri(uri);
```

Reply:
Great feedback...thanks!

---
In reply to: [278143302](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1107#discussion_r278143302) [](ancestors = 278143302)